### ISSUE TITLE
Cursor Pagination not returning expected results

### ISSUE THREAD

**Issue by defrex** [prisma/prisma#3505]
## Bug description

I'm using cursor-based pagination, and have run into a scenario where it's not returning the results I would expect.

## How to reproduce

Here are two queries, one without pagination, and one with.

```typescript
prisma.tasks.findMany({
  where: { child: { id: 136 } },
  orderBy: [{ confirmedAt: "desc" }, { id: "asc" }],
})

// Results:
[
  {
    id: 176,
    childId: 136,
    createdAt: 2020-09-02T17:58:31.048Z,
    updatedAt: 2020-09-02T17:58:31.048Z,
    completedAt: 2020-09-02T17:58:31.046Z,
    confirmedAt: null,
  },
  {
    id: 177,
    childId: 136,
    createdAt: 2020-09-02T17:58:31.058Z,
    updatedAt: 2020-09-02T17:58:31.058Z,
    completedAt: null,
    confirmedAt: null,
  },
  {
    id: 178,
    childId: 136,
    createdAt: 2020-09-02T17:58:31.067Z,
    updatedAt: 2020-09-02T17:58:31.067Z,
    completedAt: null,
    confirmedAt: null,
  },
  {
    id: 179,
    childId: 136,
    createdAt: 2020-09-02T17:58:31.075Z,
    updatedAt: 2020-09-02T17:58:31.075Z,
    completedAt: 2020-09-02T17:58:31.074Z,
    confirmedAt: null,
  }
]
```

```typescript
prisma.tasks.findMany({
  where: { child: { id: 136 } },
  orderBy: [{ confirmedAt: "desc" }, { id: "asc" }],
  take: 2,
  cursor: { id: 177 },
  skip: 1,
})

// Results:
[]
```

## Expected behavior

I would expect the paginated query to return that last two results.

```typescript
[
  {
    id: 178,
    childId: 136,
    createdAt: 2020-09-02T17:58:31.067Z,
    updatedAt: 2020-09-02T17:58:31.067Z,
    completedAt: null,
    confirmedAt: null,
  },
  {
    id: 179,
    childId: 136,
    createdAt: 2020-09-02T17:58:31.075Z,
    updatedAt: 2020-09-02T17:58:31.075Z,
    completedAt: 2020-09-02T17:58:31.074Z,
    confirmedAt: null,
  }
]
```

## Environment & setup

- OS: macOS
- Database: PostgreSQL
- Node.js version: v12.18.1
- Prisma version:
```
@prisma/cli          : 2.5.1
Current platform     : darwin
Query Engine         : query-engine c88925ce44a9b89b4351aec85ba6a28979d2658e (at node_modules/@prisma/cli/query-engine-darwin)
Migration Engine     : migration-engine-cli c88925ce44a9b89b4351aec85ba6a28979d2658e (at node_modules/@prisma/cli/migration-engine-darwin)
Introspection Engine : introspection-core c88925ce44a9b89b4351aec85ba6a28979d2658e (at node_modules/@prisma/cli/introspection-engine-darwin)
Format Binary        : prisma-fmt c88925ce44a9b89b4351aec85ba6a28979d2658e (at node_modules/@prisma/cli/prisma-fmt-darwin)
Studio               : 0.261.0
```

**Comment by pantharshit00** [comment]
Hey @defrex 

Can you please the task model's schema so that I can easily try this locally? Thanks!

**Comment by defrex** [comment]
Sure thing! I redacted a few of the fields from the example for easier reading, but their values shouldn't make any difference.

```prisma
model Task {
  @@map(name: "task")

  id             Int             @default(autoincrement()) @id
  createdAt      DateTime        @default(now())
  updatedAt      DateTime        @default(now())
  amount         Int             @default(0)
  archivedAt     DateTime?
  childId        Int
  completedAt    DateTime?
  confirmedAt    DateTime?
  paidAt         DateTime?
  parentId       Int
  recurrenceId   Int?
  title          String
  validAt        DateTime?
  expiresAt      DateTime?
  child          User            @relation("task_childIdTouser", fields: [childId], references: [id])
  parent         User            @relation("task_parentIdTouser", fields: [parentId], references: [id])
  taskRecurrence TaskRecurrence? @relation(fields: [recurrenceId], references: [id])
}
```

**Comment by pantharshit00** [comment]
Hey @defrex

I am unable to reproduce this. I got the expected result:
![image](https://user-images.githubusercontent.com/22195362/92071901-94dfe080-edcd-11ea-895d-c3672a81f7bb.png)


Can you please try this again with the latest release(2.6) ? Also, if this is still reproducible after the update, please share your Child model as well.

**Comment by defrex** [comment]
@pantharshit00 working on a repro with the latest version.

However, fwiw, in my example `confirmedAt` was `null` for all instances but has a value in all of yours. That may impact the outcome.

**Comment by defrex** [comment]
Confirmed the issue is still happening in `2.6.1`. I also got 2 test regressions from upgrading to `2.6.1`, but perhaps that's a topic for another issue.

**Comment by defrex** [comment]
@pantharshit00 here is a minimal reproduction.

https://github.com/defrex/prisma-page-repro

**Comment by jshimko** [comment]
I'm having the same issue and have narrowed it down to my query having an `orderBy` item that is potentially null.  For example, I have products and vendors.  Third party vendors are an optional relation on products.

```prisma
model Product {
  id                          String            @default(cuid()) @id
  sku                         String            @unique
  name                        String?
  price                       Int
  vendorId                    String?
  vendor                      Vendor?           @relation(fields: [vendorId], references: [id])
  createdAt                   DateTime          @default(now())
  updatedAt                   DateTime          @updatedAt
}

model Vendor {
  id           String         @default(cuid()) @id
  name         String
  products     Product[]
  createdAt    DateTime       @default(now())
  updatedAt    DateTime       @updatedAt
}
```

On our catalog page, we want to favor internal products over third party vendor products in the search results, so we've been sorting by `vendorId` with `orderBy: [{ vendorId: 'desc' }]`.  This makes all products with a null `vendorId` show up first in the results.  However, as soon as I add a `cursor` arg to skip to the next page, it returns 0 results even though we have thousands in the db.  Full example query...

Confirm sanity...

```js
// returns 10124
await prisma.product.count();
```

Get the first 10 results (works as expected with null vendorId's first)
```js
await prisma.product.findMany({
  orderBy: [
    { vendorId: 'desc' },
    { createdAt: 'desc' }
  ],
  take: 10
});
```
Now when I try to fetch the next page, I get zero results even though there are still ~10k products left matching the query.
```js
// returns 0 results
await prisma.product.findMany({
  orderBy: [
    { vendorId: 'desc' },
    { createdAt: 'desc' }
  ],
  take: 10,
  skip: 1,
  cursor: {
    id: '5cb0aac668e4040064000dc2' // id of tenth result above
  }
});
```
And here's the only clue I've come across... if I remove the `{ vendorId: 'desc' }` from the `orderBy` array, pagination starts working as expected again.  So this works...
```js
await prisma.product.findMany({
  orderBy: [
    { createdAt: 'desc' }
  ],
  take: 10,
  skip: 1,
  cursor: {
    id: '5cb0aac668e4040064000dc2' // id of tenth result above
  }
});
```

Also, if I add any other non-null field to the `orderBy` array, it works as expected.  So it appears to be something to do with the combination of sorting on a null field while also using cursor pagination.  For example, replacing the vendorId sort with price instead works...

```js
// returns the next 10 results as expected
await prisma.product.findMany({
  orderBy: [
    { price: 'asc' },
    { createdAt: 'desc' }
  ],
  take: 10,
  skip: 1,
  cursor: {
    id: '5cb0aac668e4040064000dc2'
  }
});
```

I'm not sure where this got introduced, but it was some time in the last few releases because this was working on our production site a few weeks ago.  I'm assuming this started somewhere around when the support for multiple `orderBy` entries was added.  

Any recommendations?

**Comment by pantharshit00** [comment]
I can reproduce this now with @defrex's reproduction: https://github.com/defrex/prisma-page-repro


Thanks @jshimko for the write up as well. We will need to fix this.

**Comment by dpetrick** [comment]
Internal note: Pinned the issue to originate form incorrect SQL generation: 
```
`Task`.`completedAt` = `order_cmp`.`completedAt`
```
The above will disregard `NULL` values because `NULL != NULL`. Issue fix surface is limited to this specific query, probably not too hard to fix.

**Comment by dpetrick** [comment]
Fix available from `2.8.0-dev.13` onwards (will be in the 2.8 release).
